cmd/bosun/expr/funcs.go

```diff
@@ -574,39 +618,122 @@ func Band(e *State, T miniprofiler.Timer, query, duration, period string, num fl
 				if e.squelched(res.Tags) {
 					continue
 				}
-				newarr := true
-				for _, a := range r.Results {
-					if !a.Group.Equal(res.Tags) {
-						continue
-					}
-					newarr = false
-					values := a.Value.(Series)
-					for k, v := range res.DPS {
-						i, e := strconv.ParseInt(k, 10, 64)
-						if e != nil {
-							err = e
-							return
-						}
-						values[time.Unix(i, 0).UTC()] = float64(v)
-					}
+				if err = rfunc(r, res); err != nil {
+					return
 				}
-				if newarr {
-					values := make(Series)
-					a := &Result{Group: res.Tags}
-					for k, v := range res.DPS {
-						i, e := strconv.ParseInt(k, 10, 64)
-						if e != nil {
-							err = e
-							return
-						}
-						values[time.Unix(i, 0).UTC()] = float64(v)
-					}
-					a.Value = values
-					r.Results = append(r.Results, a)
+			}
+		}
+	})
+	return
+}
+
+func Window(e *State, T miniprofiler.Timer, query, duration, period string, num float64, rfunc string) (r *Results, err error) {
+	fn, ok := e.GetFunction(rfunc)
+	if !ok {
+		return nil, fmt.Errorf("expr: Window: no %v function", rfunc)
+	}
+	f := reflect.ValueOf(fn.F)
+	r, err = bandTSDB(e, T, query, duration, period, num, func(r *Results, res *opentsdb.Response) error {
```

Comment:
This function is kinda big to be passed in the argument list. Can we not make it anonymous?